scripts/bootstrap_flux.sh

```diff
@@ -0,0 +1,216 @@
+EXIT_COUNTER=0
+MAX_COUNTER=20
+while [ ! -f $FLUX_SERVER ]; do
+  sleep 5s
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] $FLUX_SERVER does not exist yet."
+  exit_counter=$((EXIT_COUNTER + 1))
+  if [ "$EXIT_COUNTER" -eq "$MAX_COUNTER" ]; then
+    echo "[$(date +'%m%d%Y-%T')@$(hostname)] Timeout: Failed to find file (${FLUX_SERVER})."
+    exit 1
+  fi
+done
+
+export FLUX_URI=$(cat $FLUX_SERVER)
+echo "[$(date +'%m%d%Y-%T')@$(hostname)] You can run: export FLUX_URI=$(cat $FLUX_SERVER)"
+
+if [[ "$(flux getattr size)" -eq "${FLUX_NODES}" ]]; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Flux launch successful with $(flux getattr size) nodes"
+else
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: Requested nodes=${FLUX_NODES} but Flux allocation size=$(flux getattr size)"
+  exit 1
+fi
+
+# Read configuration file with number of nodes/cores for each sub allocations
+NODES_PHYSICS=$(jq ".physics.nodes" $AMS_JSON)
+NODES_ML=$(jq ".ml.nodes" $AMS_JSON)
+NODES_CONTAINERS=$(jq ".containers.nodes" $AMS_JSON)
+re='^[0-9]+$'
+if ! [[ $NODES_PHYSICS =~ $re && $NODES_ML =~ $re && $NODES_CONTAINERS =~ $re ]] ; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: number of nodes is not an integer\
+  (${NODES_PHYSICS}, ${NODES_ML}, ${NODES_CONTAINERS})"
+  exit 1
+fi
+
+CORES_PHYSICS=$(jq ".physics.cores" $AMS_JSON)
+CORES_ML=$(jq ".ml.cores" $AMS_JSON)
+CORES_CONTAINERS=$(jq ".containers.cores" $AMS_JSON)
+if ! [[ $CORES_PHYSICS =~ $re && $CORES_ML =~ $re && $CORES_CONTAINERS =~ $re ]] ; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: number of cores is not an integer\
+  (${CORES_PHYSICS}, ${CORES_ML}, ${CORES_CONTAINERS})"
+fi
+
+GPUS_PHYSICS=$(jq ".physics.gpus" $AMS_JSON)
+GPUS_ML=$(jq ".ml.gpus" $AMS_JSON)
+GPUS_CONTAINERS=$(jq ".containers.gpus" $AMS_JSON)
+if ! [[ $GPUS_PHYSICS =~ $re && $GPUS_ML =~ $re && $CORES_CONTAINERS =~ $re ]] ; then
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Error: number of GPUs is not an integer\
+  (${GPUS_PHYSICS}, ${GPUS_ML}, ${GPUS_CONTAINERS})"
+  exit 1
+fi
+
+# Partition resources for physics, ML and containers (RabbitMQ, filtering)
+# NOTE: with more recent Flux (>=0.46), we could use flux alloc --bg instead
+JOBID_PHYSICS=$(
+  flux mini batch --job-name="ams-physics" \
+    --output="ams-physics-{{id}}.log" \
+    --exclusive \
+    --nslots=1 --nodes=$NODES_PHYSICS \
+    --cores-per-slot=$CORES_PHYSICS \
+    --gpus-per-slot=$GPUS_PHYSICS \
+    --wrap sleep inf
+)
+sleep 2s
+
+# NOTE: with more recent versions of Flux, instead of sed here we could use flux jobs --no-header
+if [[ "$(flux jobs -o '{status}' $JOBID_PHYSICS | sed -n '1!p')" == "RUN" ]]; then
+  FLUX_PHYSICS_URI=$(flux uri --remote $JOBID_PHYSICS)
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Physics batch job ($JOBID_PHYSICS)\
+  started with nodes=${NODES_PHYSICS}, cores=${CORES_PHYSICS}, gpus=${GPUS_PHYSICS}"
+else
+  echo "[$(date +'%m%d%Y-%T')@$(hostname)] Warning: failed to launch physics batch job ($JOBID_PHYSICS)"
+fi
+
```

Comment:
@lpottier Create a function that does this. The signature of the function should get (I guess)
NODES, CORES, GPUS, job-name etc.

Reply:
You can then call it every time you try to start an sleep job